### PR TITLE
Chore/feeds test suite cleanup

### PR DESCRIPTION
**Pull request by Oozyx**

### CODE REVIEW

File: contracts/src/v0.8/feeds/test/DualAggregator.t.sol [hunk 1]
```diff
@@ -345,57 +339,109 @@ contract SetConfig is DualAggregatorBaseTest {
...
+      onchainConfig,
+      offchainConfigVersion,
+      offchainConfig
+    );
+
+    uint32 blockNumber = uint32(block.number);
+
+    vm.expectEmit();
+    emit ConfigSet(0, configDigest, 1, signers, transmitters, f, onchainConfig, offchainConfigVersion, offchainConfig);
+    s_aggregator.setConfig(signers, transmitters, f, onchainConfig, offchainConfigVersion, offchainConfig);
```
Comment (eduard-cl) [inline contracts/src/v0.8/feeds/test/DualAggregator.t.sol:394]: Shouldn't the `vm.expectEmit()` be just before the `setConfig()`?

File: contracts/src/v0.8/feeds/test/DualAggregator.t.sol [hunk 2]
```diff
@@ -345,57 +339,109 @@ contract SetConfig is DualAggregatorBaseTest {
...
+      newConfigDigest,
+      2,
+      newSigners,
+      newTransmitters,
+      f,
+      onchainConfig,
+      offchainConfigVersion,
+      offchainConfig
+    );
+    s_aggregator.setConfig(newSigners, newTransmitters, f, onchainConfig, offchainConfigVersion, offchainConfig);
```
Comment (eduard-cl) [inline contracts/src/v0.8/feeds/test/DualAggregator.t.sol:428]: same here, comment above

File: contracts/src/v0.8/feeds/test/DualAggregator.t.sol [hunk 3]
```diff
@@ -514,14 +550,14 @@ contract Transmit is ConfiguredDualAggregatorBaseTest {
...
     ss[0] = bytes32(abi.encodePacked("1"));
 
-    aggregator.transmit(reportContext, report, rs, ss, rawVs);
+    s_aggregator.transmit(reportContext, report, rs, ss, rawVs);
   }
 
   function test_RevertIf_CalldataLengthMismatch() public {
-    vm.startPrank(transmitters[0]);
+    vm.startPrank(s_transmitters[0]);
     vm.expectRevert(DualAggregator.CalldataLengthMismatch.selector);
```
Comment (eduard-cl) [inline contracts/src/v0.8/feeds/test/DualAggregator.t.sol]: Similar to what I said in the other comments, following the solidity guide:

> There should be no code between `vm.expectEmit`/`vm.expectRevert` and the function call.

File: contracts/src/v0.8/feeds/test/DualAggregator.t.sol [hunk 4]
```diff
@@ -1968,14 +1969,14 @@ contract WithdrawFunds is ConfiguredDualAggregatorBaseTest {
...
-    aggregator.withdrawFunds(USER, 42);
+    s_aggregator.withdrawFunds(USER, 42);
   }
 
   // TODO: need to run a transmit to ensure the user has a lot to withdraw
   // function test_RevertIf_InsufficientBalance() public {
   //   vm.expectRevert("insufficient balance");
   //
-  //   aggregator.withdrawFunds(USER, 1e9);
+  //   s_aggregator.withdrawFunds(USER, 1e9);
```
Comment (eduard-cl) [inline contracts/src/v0.8/feeds/test/DualAggregator.t.sol]: Now that we have the transmits implemented, could you add this test case?

**Review by eduard-cl — COMMENTED** [review]
Overall looks good, left some small comments

File: contracts/src/v0.8/feeds/test/DualAggregator.t.sol (hunk 1 above)
Comment (Oozyx) [reply contracts/src/v0.8/feeds/test/DualAggregator.t.sol:394]: https://book.getfoundry.sh/cheatcodes/expect-emit?highlight=expectemit#expectemit

File: contracts/src/v0.8/feeds/test/DualAggregator.t.sol (hunk 3 above)
Comment (Oozyx) [reply contracts/src/v0.8/feeds/test/DualAggregator.t.sol]: here you're correct, we should place expectRevert above the function call

File: contracts/src/v0.8/feeds/test/DualAggregator.t.sol (hunk 1 above)
Comment (eduard-cl) [reply contracts/src/v0.8/feeds/test/DualAggregator.t.sol:394]: I see, thank you!

File: contracts/src/v0.8/feeds/test/DualAggregator.t.sol (hunk 2 above)
Comment (eduard-cl) [reply contracts/src/v0.8/feeds/test/DualAggregator.t.sol:428]: Resolved, conversation above, we need to declare the emit event we expect.

File: contracts/src/v0.8/feeds/test/DualAggregator.t.sol (hunk 4 above)
Comment (Oozyx) [reply contracts/src/v0.8/feeds/test/DualAggregator.t.sol]: done

**Review by eduard-cl — APPROVED** [review]

**Pull request merged**